src/plugins/data/public/index_patterns/index_patterns_service.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  UiSettingsClientContract,
+  SavedObjectsClientContract,
+  HttpServiceBase,
+  NotificationsStart,
+} from 'src/core/public';
+import { FieldFormatsStart } from '../field_formats_provider';
+import { Field, FieldList } from './fields';
+import { setNotifications, setFieldFormats } from './services';
+
+import { createFlattenHitWrapper, formatHitProvider, IndexPatterns } from './index_patterns';
+
+export interface IndexPatternDependencies {
+  uiSettings: UiSettingsClientContract;
+  savedObjectsClient: SavedObjectsClientContract;
+  http: HttpServiceBase;
+  notifications: NotificationsStart;
+  fieldFormats: FieldFormatsStart;
+}
+
+/**
+ * Index Patterns Service
+ *
+ * @internal
+ */
+export class IndexPatternsService {
```

Comment:
(a) please move the code rather then copy it.
(b) the most important part of this PR is to initialize and expose the `indexpattern` service from `src/plugins/data/public/plugin.ts`
(c) then re-export whatever is needed via `src/legacy/core_plugins/data/public/plugin.ts`

Reply:
Done